lib/ain-evm/src/transaction/mod.rs

```diff
@@ -300,6 +300,10 @@ impl SignedTx {
             TransactionV2::EIP1559(tx) => tx.hash(),
         }
     }
+
+    pub fn get_tx_type(&self) -> U256 {
+        U256::from(EnvelopedEncodable::type_id(&self.transaction).unwrap_or_default())
```

Comment:
Is it safe to use `unwrap_or_default` here? Maybe this should return a `Result` instead.

Reply:
This is correct. For legacy TransactionV2 txs, None is returned. Thus we unwrap to default (0 value), and Legacy txs are of type value 0.